google-cloud-firestore/src/main/java/com/google/cloud/firestore/TransactionOptions.java

```diff
@@ -16,65 +16,98 @@
 
 package com.google.cloud.firestore;
 
+import com.google.api.core.InternalApi;
 import com.google.common.base.Preconditions;
+import com.google.firestore.v1.TransactionOptions.ReadOnly;
+import com.google.firestore.v1.TransactionOptions.ReadWrite;
+import com.google.protobuf.Timestamp;
+import com.google.protobuf.TimestampOrBuilder;
 import java.util.concurrent.Executor;
 import javax.annotation.Nonnull;
 import javax.annotation.Nullable;
 
 /** Options specifying the behavior of Firestore Transactions. */
 public final class TransactionOptions {
 
+  private static final TransactionOptions DEFAULT_READ_WRITE_TRANSACTION_OPTIONS =
+      readWriteOptionsBuilder().build();
+
   private static final int DEFAULT_NUM_ATTEMPTS = 5;
 
-  private final int numberOfAttempts;
   private final Executor executor;
+  private final EitherReadOnlyOrReadWrite options;
 
-  TransactionOptions(int maxAttempts, Executor executor) {
-    this.numberOfAttempts = maxAttempts;
+  TransactionOptions(Executor executor, EitherReadOnlyOrReadWrite options) {
     this.executor = executor;
+    this.options = options;
   }
 
+  /**
+   * @return int Representing the initial number of attempts a read-write transaction will be
+   *     attempted
+   * @deprecated As of v2.0.0, only applicable to Read-Write transactions. Use {@link
```

Comment:
I was once told that we should use lowercase here.

Reply:
Updated throughout